session_3/blogspot/app/controllers/articles_controller.rb

```diff
@@ -0,0 +1,78 @@
+class ArticlesController < ApplicationController
+  load_and_authorize_resource
+  
+  before_action :set_article, only: [:show, :edit, :update, :destroy]
+
+  # GET /articles
+  # GET /articles.json
+  def index
+    @articles = Article.all
+  end
+
+  # GET /articles/1
+  # GET /articles/1.json
+  def show
+    if !@article.public && !current_user.admin && @article.user_id != session[:user_id] && session[:private_articles_remaining]>0
+      session[:private_articles_remaining] -= 1
+    elsif !@article.public && session[:private_articles_remaining] < 1
+      redirect_to root_url, notice: "Private Articles Limit Exceeded"
```

Comment:
`alert` instead of `notice` would be more appropriate.

Reply:
Ohk,.Thanks for the info :+1: